mesonbuild/linkers/detect.py

```diff
@@ -151,7 +151,7 @@ def guess_nix_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
     _, o, e = Popen_safe(compiler + check_args)
     v = search_version(o + e)
     linker: DynamicLinker
-    if o.startswith('LLD'):
+    if re.search(r'^(\S+\s+)*LLD', o):
```

Comment:
This seems equivalent to `if 'LLD' in o`, which is how we handle GNU below.

Reply:
The idea was to only match the first line but yeah I am not sure it will work. Should have been `.*\bLLD` since period does not match newline.

I won't be able to adjust the PR over the weekend though, sorry.

Reply:
If you really want to just match the first line, I still suspect `'LLD' in o.split()[0]` will still be faster than a regex